### ISSUE TITLE
Flake-y node-test in CI

### ISSUE THREAD

**Issue by runspired** [emberjs/data#6811]
One of our node-tests sometimes times out causing the builds scenario to fail

Example failure below:

 ```
15 passing (1m)
  1 failing

  1) Acceptance: generate and destroy model blueprints
       octane
         model with attrs:
     Error: Timeout of 20000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/runner/runners/2.160.1/work/1/s/packages/model/node-tests/blueprints/model-test.js)
      at tryCatcher (/Users/runner/runners/2.160.1/work/1/s/node_modules/rsvp/dist/rsvp.js:326:21)
      at invokeCallback (/Users/runner/runners/2.160.1/work/1/s/node_modules/rsvp/dist/rsvp.js:498:33)
      at publish (/Users/runner/runners/2.160.1/work/1/s/node_modules/rsvp/dist/rsvp.js:484:9)
      at flush (/Users/runner/runners/2.160.1/work/1/s/node_modules/rsvp/dist/rsvp.js:2441:7)
      at processTicksAndRejections (internal/process/task_queues.js:75:11)
```

**Comment by runspired** [comment]
Looks like it doesn't always fail for the same reason either:

```
1) Acceptance: generate and destroy model blueprints
       octane
         model-test:
     Error: EINVAL: invalid argument, readlink '/Users/runner/runners/2.160.1/work/1/s/packages/serializer/node_modules/@ember-data/serializer'
      at Object.realpathSync (fs.js:1554:9)
      at _getRealDirectoryPath (/Users/runner/runners/2.160.1/work/1/s/node_modules/resolve-package-path/lib/resolve-package-path.js:90:31)
      at resolvePackagePath.getRealDirectoryPath (/Users/runner/runners/2.160.1/work/1/s/node_modules/resolve-package-path/index.js:83:12)
      at PackageInfoCache._readPackage (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:447:20)
      at /Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:679:25
      at Array.forEach (<anonymous>)
      at PackageInfoCache._readNodeModulesList (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:641:13)
      at /Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:666:25
      at Array.forEach (<anonymous>)
      at PackageInfoCache._readNodeModulesList (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:641:13)
      at PackageInfoCache._readPackage (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:569:30)
      at /Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:679:25
      at Array.forEach (<anonymous>)
      at PackageInfoCache._readNodeModulesList (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:641:13)
      at /Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:666:25
      at Array.forEach (<anonymous>)
      at PackageInfoCache._readNodeModulesList (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:641:13)
      at PackageInfoCache._findPackage (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:377:34)
      at /Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/package-info.js:206:40
      at Array.forEach (<anonymous>)
      at PackageInfo.addDependencies (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/package-info.js:192:21)
      at /Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:286:26
      at Array.forEach (<anonymous>)
      at PackageInfoCache._resolveDependencies (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:277:18)
      at PackageInfoCache.loadProject (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/package-info-cache/index.js:197:12)
      at new Project (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/project.js:87:47)
      at Function.closestSync (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/project.js:697:12)
      at Function.projectOrnullProject (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/models/project.js:714:22)
      at module.exports (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/cli/index.js:136:25)
      at ember (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/tests/helpers/ember.js:68:17)
      at ember (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli-blueprint-test-helpers/lib/helpers/ember.js:6:48)
      at module.exports (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli-blueprint-test-helpers/lib/ember-destroy.js:27:10)
      at /Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli-blueprint-test-helpers/lib/ember-generate-destroy.js:35:17
      at tryCatcher (/Users/runner/runners/2.160.1/work/1/s/node_modules/rsvp/dist/rsvp.js:326:21)
      at invokeCallback (/Users/runner/runners/2.160.1/work/1/s/node_modules/rsvp/dist/rsvp.js:498:33)
      at publish (/Users/runner/runners/2.160.1/work/1/s/node_modules/rsvp/dist/rsvp.js:484:9)
      at flush (/Users/runner/runners/2.160.1/work/1/s/node_modules/rsvp/dist/rsvp.js:2441:7)
      at processTicksAndRejections (internal/process/task_queues.js:75:11)

  2) Acceptance: generate and destroy model blueprints
       octane
         "before each" hook for "model-test-test foo":
     Error: process already captured
      at Object.capture (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/utilities/will-interrupt-process.js:24:13)
      at module.exports (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/lib/cli/index.js:74:24)
      at ember (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli/tests/helpers/ember.js:68:17)
      at ember (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli-blueprint-test-helpers/lib/helpers/ember.js:6:48)
      at module.exports (/Users/runner/runners/2.160.1/work/1/s/node_modules/ember-cli-blueprint-test-helpers/lib/ember-new.js:36:10)
      at Context.<anonymous> (node-tests/blueprints/model-test.js:164:14)
      at processImmediate (internal/timers.js:439:21)
```

**Comment by runspired** [comment]
One more note, this has only been occurring on azure where we run on this vm `'macOS-10.14'`, restarting CI often resolves the issue.

It has not been occurring on github-actions CI where we run this on the following VMs

```
      matrix:
        os: [macOS-10.14, windows-2016, ubuntu-18.04]
```

**Comment by runspired** [comment]
```
2019-11-27T21:03:47.7849920Z git-lfs/2.9.0 (GitHub; linux amd64; go 1.13.1)
```

is in our log for the "macos" command and I'm wondering if the `runson` here is an accident that screws with the actual vm os choice: Possibly we aren't actually testing on a mac for github-actions?

```
  basic-tests:
    strategy:
      matrix:
        os: [macOS-10.14, windows-2016, ubuntu-18.04]
    runs-on: ubuntu-latest
```

**Comment by runspired** [comment]
It turned out we weren't running macos properly on gh-actions, this pr fixed that #6828 

We haven't seen this flakiness in either in a few days so I'm going to close this for now.
